src/hardware/setup.rs

```diff
@@ -288,23 +288,32 @@ pub fn setup(
 
     info!("Setup ADC");
 
+    let mut syscfg = device.SYSCFG;
+    let mut exti = device.EXTI;
+    let mut rdyn = gpioc.pc11.into_floating_input();
+    rdyn.make_interrupt_source(&mut syscfg);
+    rdyn.trigger_on_edge(&mut exti, Edge::Falling);
+    rdyn.enable_interrupt(&mut exti);
```

Comment:
I would trigger on level, not edge if possible.
And then in any case you want to enable the IRQ long **after** up the ADCs.

Reply:
Moved irq enable a bit now but this is still relevant.